src/Paket.Core/Installation/InstallProcess.fs

```diff
@@ -293,20 +296,31 @@ let private applyBindingRedirects isFirstGroup createNewBindingFiles cleanBindin
                 librariesForPackage
                 |> Seq.choose(fun (library,redirects,profile) ->
                     try
-                        let assembly = Mono.Cecil.AssemblyDefinition.ReadAssembly(library.Path)
-                        Some (assembly, BindingRedirects.getPublicKeyToken assembly, assembly.MainModule.AssemblyReferences, redirects, profile)
+                        let assemblyName = AssemblyName.GetAssemblyName(library.Path)
+                        let publicKeyToken = AssemblyMetadata.getPublicKeyToken assemblyName
+                        let metadataName = AssemblyMetadata.AssemblyName assemblyName
+                        let references = 
+                            match AssemblyMetadata.getAssemblyReferences metadataName with
+                            | Trial.Pass list -> list
+                            | Trial.Warn(list, warn) -> 
+                                warn |> List.iter (fun ex -> ex |> string |> traceWarn)
+                                list
+                            | Trial.Fail exns ->
+                                exns |> List.iter (fun ex -> ex |> string |> traceError)
+                                List.empty
```

Comment:
Is `List.empty` the correct think to do here?

Reply:
probably not – should I just re-throw here?

Reply:
Probably yes, it will be caught with an evil `with _ -> None` below. We probably should add some logging there (at the very least in verbose mode)

Reply:
it would get double-logged then -- but I think we can do a throw with inner exception here to keep the original info/stack and get it logged in the caller only